src/bigquery/src/client.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains the BigQuery client and related types.
+
+use std::sync::Arc;
+
+use crate::Result;
+use crate::query::Query;
+use crate::query_request::IntoPostQueryRequest;
+use bigquery_v2::client::JobService;
+use bigquery_v2::model::PostQueryRequest;
+
+/// Implements a Query client for the BigQuery API.
+///
+/// # Example
+/// ```
+/// # tokio_test::block_on(async {
+/// # use google_cloud_bigquery::client::QueryClient;
+/// let client = QueryClient::builder().build().await?;
+/// // use `client` to run queries to BigQuery.
+/// # gax::client_builder::Result::<()>::Ok(()) });
+/// ```
+///
+/// [with_endpoint()]: ClientBuilder::with_endpoint
+/// [with_credentials()]: ClientBuilder::with_credentials
+/// [Private Google Access with VPC Service Controls]: https://cloud.google.com/vpc-service-controls/docs/private-connectivity
+/// [Application Default Credentials]: https://cloud.google.com/docs/authentication#adc
+#[derive(Clone, Debug)]
+pub struct QueryClient {
+    job_service: Arc<JobService>,
+    project_id: String,
+}
+
+const DEFAULT_HOST: &str = "https://bigquery.googleapis.com";
+
+impl QueryClient {
+    /// Returns a builder for [QueryClient].
+    ///
+    /// # Example
+    /// ```
+    /// # use google_cloud_bigquery::client::QueryClient;
+    /// # async fn sample() -> anyhow::Result<()> {
+    /// let client = QueryClient::builder().build().await?;
+    /// # Ok(()) }
+    /// ```
+    pub fn builder() -> ClientBuilder {
+        ClientBuilder::new()
+    }
+
+    /// Start a QueryRequest and returns a Query handle to monitor progress.
+    pub async fn query<T>(&self, req: T) -> Result<Query>
+    where
+        T: IntoPostQueryRequest,
+    {
+        let mut req: PostQueryRequest = req.into_post_query_request();
+        if req.project_id.is_empty() {
+            req.project_id = self.project_id.clone();
+        }
+        let res = self
+            .job_service
+            .query()
+            .set_project_id(req.project_id)
+            .set_or_clear_query_request(req.query_request)
+            .send()
+            .await?;
```

Comment:
Does this service do any exponential back off / retries? If so, is there a way to configure them? The BQ API makes it difficult to rely purely on HTTP status codes.

Reply:
right now there is no retry set up, just the base one from the `gax` package. That's what I still need to explore, but the reference implementation that I have been working is in Go for the whole query experience.